pkg/controller/replicaset/replica_set.go

```diff
@@ -482,19 +481,20 @@ func (rsc *ReplicaSetController) manageReplicas(filteredPods []*v1.Pod, rs *exte
 			if err != nil {
 				// Decrement the expected number of creates because the informer won't observe this pod
 				glog.V(2).Infof("Failed creation, decrementing expectations for replica set %q/%q", rs.Namespace, rs.Name)
-				rsc.expectations.CreationObserved(rsKey)
 			}
 			return err
 		})
 
 		// any skipped pods that we never attempted to start shouldn't be expected.
-		skippedPods := diff - totalProcessedBatchSize
+		skippedPods := diff - successfulCreations
 		if skippedPods > 0 {
 			glog.V(2).Infof("Slow-start failure. Skipping creation of %d pods, decrementing expectations for replica set %q/%q", skippedPods, rs.Namespace, rs.Name)
 			for i := 0; i < skippedPods; i++ {
 				// Decrement the expected number of creates because the informer won't observe this pod
 				rsc.expectations.CreationObserved(rsKey)
 			}
+			// The skipped pods will be retried later. The next controller resync will
```

Comment:
This comment is intended to be put here (with no codes afterwards). It is to remind that the skipped pods will be retried later. Perhaps we can put it before the if statement.